pkg/storaged/partitions/partition.jsx

```diff
@@ -118,6 +141,62 @@ const PartitionCard = ({ card, block, unused_space_warning, resize_info }) => {
         return grow_dialog(client, block_part, resize_info, true);
     }
 
+    function set_type_dialog() {
+        const choices = Object.keys(type_names).map(k => ({ value: k, title: type_names[k] }));
+        choices.push({ title: _("Custom"), value: "custom" });
+
+        function validate_type(val) {
+            if (block_ptable.Type == "dos") {
+                const hex_rx = /^[a-fA-F0-9]{2}$/;
+                if (!hex_rx.test(val))
+                    return _("Type must contain exactly two hexadecimal characters (0 to 9, A to F).");
+            } else {
+                /* We let people use any 128 bit value as a UUID, even
+                   those that are not defined by RFC 4122. This is
+                   what the rest of the storage stack does as well,
+                   and the "BIOS boot" UUID is in fact invalid
+                   according to RFC 4122.
+
+                   But we do insist that the dashes are in the right
+                   place, because UDisks2 does as well.
+                */
+                const uuid_rx_1 = /^[a-fA-F0-9-]*$/;
+                if (!uuid_rx_1.test(val))
+                    return _("Type can only contain the characters 0 to 9, A to F, and \"-\".");
+                const uuid_rx_2 = /^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$/;
+                if (!uuid_rx_2.test(val))
+                    return _("Type must be of the form NNNNNNNN-NNNN-NNNN-NNNN-NNNNNNNNNNNN.");
+            }
+        }
+
+        dialog_open({
+            Title: cockpit.format(_("Set partition type of $0"), block_name(block)),
+            Fields: [
+                SelectOne("type", _("Type"),
+                          {
+                              value: (type_names[type] ? type : "custom"),
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19800-20240112-105449-b22ddb10-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)